### PR TITLE
deck: Implemented Fisher-Yates shuffling algorithm to shuffle the deck

### PR DESCRIPTION
**Pull request by sethdivyansh**
Fixes: #2

## Description
Shuffled the deck using Fiesher-Yates algorithm

## Checklist
- [ ] I have tested these changes locally.
- [ ] I have reviewed the code and ensured it follows the project's coding guidelines.
- [ ] I have updated the documentation, if necessary.
- [ ] I have assigned reviewers to this pull request.

### CODE REVIEW

**Comment by kuv2707** [comment]
@sethdivyansh Please resolve merge conflicts.

**Comment by sethdivyansh** [comment]
@kuv2707 i have resolved the conflicted.

**Comment by kuv2707** [comment]
I suggest you write a test for this in deck.test.ts
Where you just check if the array order before and after the function call to shuffle is different.


Its not exactly required, but it would give you a chance to write unit tests for your logic.

We use the 'jest' library for testing.


The test should be in the same commit.

**Comment by sethdivyansh** [comment]
@kuv2707 i have added the shuffle deck test in deck.test.ts

File: backend/tests/deck.test.ts [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
+
+        // Create a copy of the deck for comparison
+        const originalDeck = [...deck];
+        shuffle(deck);
+
+        // Check that the order of elements has changed
+        const orderChanged = deck.some(
+            (card, index) => card !== originalDeck[index]
+        );
+        expect(orderChanged).toBe(true);
```
Comment (kuv2707) [inline backend/tests/deck.test.ts:26]: Great work detecting order change!

**Review by kuv2707 — APPROVED** [review]
Nicely done!

**Comment by kuv2707** [comment]
Merged, thanks @sethdivyansh !

**Pull request merged**